src/utils/logging.jl

```diff
@@ -60,22 +60,24 @@ end
 Returns a summary of log event counts by level.
 """
 function report_log_summary(tracker::LogEventTracker)
-    text = "\nLog message summary:\n"
+    io = IOBuffer()
+    print(io, "\nLog message summary:")
     # Order by criticality.
     for level in sort!(collect(keys(tracker.events)); rev = true)
         num_events = length(tracker.events[level])
-        text *= "\n$num_events $level events:\n"
+        print(io, "\n\n", num_events, " ", level, " events:")
```

Comment:
Like, I _guess_ this is slightly more performant, and it's sylistically inoffensive, but the context is so very much not performance critical here.